File: app/src/main/java/com/moictab/curriculumvitae/controller/ProjectController.java
package com.moictab.curriculumvitae.controller;

import android.content.Context;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.moictab.curriculumvitae.model.Project;

import java.lang.reflect.Type;
import java.util.List;

/**
 * Created by moict on 16/10/2017.
 */

public class ProjectController extends BaseController {

    private final static String PROJECTS_FILENAME = "projects.json";

    public List<Project> GetProjectsFromAssets(Context context) {
        Type list = new TypeToken<List<Project>>() {
        }.getType();

        return new Gson().fromJson(GetJsonFromResources(context, PROJECTS_FILENAME), list);
    }
}

File: app/src/main/java/com/moictab/curriculumvitae/controller/ProjectController.kt
package com.moictab.curriculumvitae.controller

import android.content.Context

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.moictab.curriculumvitae.model.Project

class ProjectController : BaseController() {
  fun getProjectsFromAssets(context: Context): List<Project>? {
    val list = object : TypeToken<List<Project>>() {}.type
    return Gson().fromJson<List<Project>>(getJsonFromResources(context, PROJECTS_FILENAME), list)
  }

  companion object {
    private const val PROJECTS_FILENAME = "projects.json"
  }
}
